tests/test_test_runner.py

```diff
@@ -3,48 +3,83 @@
 
 import pytest
 
-from dojo_toolkit.test_runner import DoctestTestRunner, SubprocessTestRunner
+from dojo_toolkit.test_runner import DoctestTestRunner
 
 MODULE_DIR = os.path.dirname(os.path.abspath(__file__))
 
 
 @pytest.fixture
-def code_path():
-    return "/path/to/my/code"
-
-
-class MockTestRunner(SubprocessTestRunner):
-    cmd = "echo"
-
-
-@pytest.fixture
-def mock_test_runner(code_path):
-    return MockTestRunner(code_path, sound_player=mock.Mock())
-
-
-class WrongTestRunner(DoctestTestRunner):
-    cmd = "parangaricutirimicuaro"
-
-
-@pytest.fixture
-def wrong_test_runner(code_path):
-    with mock.patch('dojo_toolkit.test_runner.notifier'):
-        return WrongTestRunner(code_path, sound_player=mock.Mock())
+def code_file(tmpdir):
+    code_file = tmpdir.join('foo.py')
+    return code_file
 
 
 @mock.patch('dojo_toolkit.test_runner.notifier')
-def test_mock(notifier, mock_test_runner, code_path):
-    """
-    test SubprocessTestRunner when cmd not fail
-    """
-    assert 'echo' in mock_test_runner.cmd
-    assert mock_test_runner.run()
+def test_doctest_test_runner_real_file_cmd_success(notifier, code_file):
+    sound_player_mock = mock.Mock()
+    code = [
+        '"""',
+        '>>> 1 + 2',
+        '3',
+        '"""',
+    ]
+    code_file.write('\n'.join(code))
+    test_runner = DoctestTestRunner(code_path=str(code_file), sound_player=sound_player_mock)
```

Comment:
```suggestion
    test_runner = DoctestTestRunner(code_path=str(code_file.dirpath()), sound_player=sound_player_mock)
```

Reply:
`code_path` receives the directory path for the file, then the `DoctestTestRunner` run the doctest for every file inside that directory (in this case, the `foo.py`)